### DISCUSSION TITLE
How to handle arrays as parameters with useContractWrite?

### DISCUSSION THREAD

**Discussion by Harry180** [wevm/wagmi#2717] [Q&A]
I'm trying to figure out how to handle more complex data types than uint or boolean arguments to pass to smart contract function.

Been stuck with it

Here is my configuration:
``` const { config } = usePrepareContractWrite({
        address: contractAddress,
        abi: [
            {
                name: 'batchAction',
                type: 'function',
                stateMutability: 'nonpayable',
                inputs: [{internalType: 'address[]', name: 'arrayOfAddresses', type: 'address[]'},{internalType: 'uint256[]', name: 'arrayOfNumbers', type: 'uint256[]'}],
                outputs: []
            },
        ],
        functionName: 'batchAction',
    })
    const { data, write } = useContractWrite(config);
    
    const addresses = ['0xaddress1', '0xaddress2'];
    const arrayOfNumbers = [1, 3];
    
    write?.({args: [addresses, arrayOfNumbers]});
    ```
    
    With use of ethers library it looks like that:
    ``` const abi= `[{
      "inputs": [
        {
          "internalType": "address[]",
          "name": "arrayOfAddresses",
          "type": "address[]"
        },
        {
          "internalType": "uint256[]",
          "name": "arrayOfNumbers",
          "type": "uint256[]"
        }
      ],
      "name": "batchAction",
      "outputs": [],
      "stateMutability": "nonpayable",
      "type": "function"
    }]`;
    
    
   const provider = new ethers.providers.Web3Provider(window.ethereum);
   await provider.send("eth_requestAccounts", []);
   const signer = provider.getSigner();
   const erc721 = new ethers.Contract(contractAddress, abi, signer);
   
   const addresses = ['0xaddress1', '0xaddress2'];
   const arrayOfNumbers = [1, 3];
   
   await erc721.batchAction(addresses,arrayOfNumbers );
   ```
   
   I feel like missing some small part of solution which cannot find on docs.

**Comment by danialzadrafiee** [comment]
I have same question there is no data for array types in viem doucmention
